RESPONSIBILITIES.md

```diff
@@ -56,6 +57,17 @@ It's our responsibility to ensure the content and code in pull requests are corr
 - If a pull request is valuable but isn't gaining traction, consider reaching out to fulfill the necessary requirements. This way, the pull request can be merged, even if the work is done by several individuals.
 - Lastly, strive for progress, not perfection.
 
+### Merging a Pull Request 
+
+It is important that commit messages are helpful in understanding the reasons for a given commit and maintain good commit hygiene by only keeping the relevant information. 
+
+Most repositories in [opensearch-project](https://github.com/opensearch-project) are configured to require commits to be squashed into a single commit when merging pull requests. This process needs human intervention to produce high quality commit messages. When merging pull requests, edit merge messages by following these steps as much as possible:
+
+- The commit subject should be concise and clearly convey what is being merged.
+- The commit body should include the details (if any) about the commit, typically inline with the PR description.
+- The commit body should include the 'Signed-Off-By:*' for all committers involved in the change.
+- There need to be a matching 'Signed-Off-By:' line for the `This commit will be authored by *` email address otherwise backport DCO checks will fail.
```

Comment:
The section you linked already mentions this as contributing guidelines:

```
The DCO is a declaration attached to every contribution made by every developer. In the commit message of the contribution, the developer simply adds a Signed-off-by statement and thereby agrees to the DCO, which you can find below or at [DeveloperCertificate.org](http://developercertificate.org/).
```


Also it is also mentioned that all workflows in opensearch-project have DCO checks enabled:

```
We require that every contribution to OpenSearch is signed with a Developer Certificate of Origin. DCO checks are enabled via a [DCO workflow app](https://github.com/apps/dco) across the entire opensearch-project organization, and your PR will fail CI without it.
```

So I'll just update it very briefly that the `'signed-off by:' implies that the contributor agrees to DCO<link>` to avoid it redundancy.